lib/engine-jit/src/engine.rs

```diff
@@ -299,4 +310,9 @@ impl JITEngineInner {
     pub fn signatures(&self) -> &SignatureRegistry {
         &self.signatures
     }
+
+    /// Shared func metadata registry.
+    pub(crate) fn func_data(&self) -> &Arc<FuncDataRegistry> {
```

Comment:
Yeah, I think we use this pattern in the trait already is why I went with it (to be more consistent), returning `&Arc<T>` is more efficient (caller gets to decide if they need to do the allocation and atomic increment to make a new Arc; they don't have to if they just want to access the internal type) but also more brittle from an API perspective because it means that this type must always hold on to an Arc internally that lives long enough. It's definitely a trade-off 🤷